src/Button/Button.js

```diff
@@ -0,0 +1,339 @@
+// @flow
+import * as React from "react";
+import defaultTokens from "@kiwicom/orbit-design-tokens";
+import styled from "styled-components";
+
+import { iconSizes } from "../Icon";
+
+export const TYPE_OPTIONS = {
+  PRIMARY: "primary",
+  SECONDARY: "secondary",
+  INFO: "info",
+  SUCCESS: "success",
+  WARNING: "warning",
+  CRITICAL: "critical",
+  FACEBOOK: "facebook",
+  GOOGLE: "google",
+};
+
+export const SIZE_OPTIONS = {
+  SMALL: "small",
+  NORMAL: "normal",
+  LARGE: "large",
+};
+
+type Props = {|
+  children?: React.Node,
+  component: string | React.Node,
+  href?: string,
+  onClick?: (e: SyntheticEvent<HTMLButtonElement>) => void,
+  external: boolean,
+  bordered: boolean,
+  disabled: boolean,
+  block: boolean,
+  type: $Values<typeof TYPE_OPTIONS>,
+  size: $Values<typeof SIZE_OPTIONS>,
+  width: number,
+  icon?: React.Node,
+  theme: typeof defaultTokens,
+|};
+
+const IconContainer = styled(({ theme, tokens, bordered, sizeIcon, type, onlyIcon, ...props }) => (
```

Comment:
I think there isn't available any other props

why there you need it ?

```
<IconContainer
          size={size}
          onlyIcon={onlyIcon}
          bordered={bordered}
          sizeIcon={sizeIcon}
          tokens={tokens}
          type={type}
          theme={theme}
        >
```


just render `<div />` without props

Reply:
We need to pass at least `children` and `className` to this div.

Reply:
OK, so i think this is more readable, and you shouldn't destruct unused props

![image](https://user-images.githubusercontent.com/7850592/41584670-2f52652c-73a8-11e8-98b7-cc821b82962c.png)